woocommerce/woocommerce
# Apply Rector suggestions for PHP 8.1

### Changes proposed in this Pull Request:

Applied suggestions based on Rector's static analysis for PHP 8.1. 

This is a continuation of https://github.com/woocommerce/woocommerce/pull/40801

Ref: D128479-code



### How to test the changes in this Pull Request:

- `plugins/woocommerce/includes/admin/class-wc-admin-importers.php`
    - Add this code snippet to your `functions.php` file or use the [[Code Snippets extension](https://pe.wordpress.org/plugins/code-snippets/)](https://pe.wordpress.org/plugins/code-snippets/), in order to decrease the batch size.
        ```bash
        add_filter('woocommerce_product_import_batch_size', function ($size) {
        	return 1;
        });
        ```
    - Create 2 simple products
    - Go to Products > All Products > Export. Click “Generate CV”
    - Go to Products > All Products > Import. And upload the .csv file from the previous step then run the importer.
    - It should complete without issues. (You might encounter this issue #44784, but it's unrelated to this PR.)
- `plugins/woocommerce/includes/admin/class-wc-admin-webhooks.php`
    - Go to WooCommerce > Settings > Advanced > Webhooks
    - Create a webhook.
    - Click “Delete permanently”
    - It should delete the webhook without issues.
- `plugins/woocommerce/includes/admin/helper/class-wc-helper.php`
    - Add the following snippet using the Code Snippets plugin:
        ```bash
        add_filter(
        'pre_option_woocommerce_feature_marketplace_enabled',
        fn() => 'no'
        );
        ```
    - Go to [https://your.site/wp-admin/admin.php?page=wc-addons](https://wc.jurassic.tube/wp-admin/admin.php?page=wc-addons)
    - Go to “My Subscriptions”
    - Click “Connect” to connect using your A8C account.
    - It should show the subscriptions page without issues.
- `plugins/woocommerce/includes/admin/importers/class-wc-tax-rate-importer.php`
    - Go to WooCommerce > Settings. Check “Enable tax rates and calculations” then save changes.
    - Go to WooCommerce > Settings > Tax > Standard rates.
    - Click “Insert row” then “Export CSV”
    - Click “Import CSV” and upload the exported file. It should import taxes without issues.
- `plugins/woocommerce/includes/admin/importers/views/html-csv-import-done.php`
    - Go to Products. Having at least on product created, then click “Export”. Then click “Generate CV”
    - Go to Products. Click “Import” then upload the .csv file from the previous step.
    - Run the importer. It should finish without issues. Clicking “View import log” should also work.
    - Notice the product was probably skipped because it existed already. That’s expected.
- `plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-coupons.php`
    - Go to Marketing > Coupons.
    - Add a new coupon. Click “Usage restriction”. Under “Products” choose an existing product then click “publish”
    - Go to Marketing > Coupons. You should see the table of coupons. The Product ID should be in the “Product IDs” column.
- `plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php`
    - Go to Products > Add New.
    - In the product type dropdown select “Variable product”
    - Click the “Variations” tab. Should be able to see the empty view without issues.
- `plugins/woocommerce/includes/admin/reports/class-wc-report-sales-by-category.php`
    - Create a product. Add a price and assign a category to the product.
    - Create an order. Set status as “Completed”. Add product from previous step as item.
    - Go to WooCommerce > Reports > Sales by Category.
    - Choose the category from the first step. Click “show”
    - Should be able to see the chart without issues.
- `plugins/woocommerce/includes/admin/settings/class-wc-settings-tax.php`
    - The tax tables should be visible, can insert or remove tax rates without issues.
    - The method `save_tax_classes` is not used anymore. Tax are saved via ajax now.
- `plugins/woocommerce/includes/admin/views/html-admin-page-status-report.php`
    - Go to WooCommerce > Status. The System status report should be visible without issues.
- `plugins/woocommerce/includes/class-wc-ajax.php`
    - Go to Products > Categories.
    - Sorting categories using the ≡ button should work as expected.
- `plugins/woocommerce/includes/class-wc-deprecated-action-hooks.php`
    - No need to test changes in this file as they are deprecated hooks that were never called anyway.
- `plugins/woocommerce/includes/class-wc-geo-ip.php`
    - No need to test this as changes in this file are insignificant. Will never break the app.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Apply Rector suggestions for PHP 8.1
#### Comment 

</details>
